### PR TITLE
add export multichannel to kaldi

### PR DESCRIPTION
**Pull request by jtrmal**
similar to the POC code but now in the correct branch and resolved some minuscule changes

### CODE REVIEW

File: lhotse/kaldi.py [hunk 1]
```diff
@@ -207,26 +209,30 @@ def export_to_kaldi(
...
+    # the recording <-> supervision information.
     cuts = CutSet.from_manifests(
         recordings=recordings, supervisions=supervisions
     ).trim_to_supervisions()
 
     # wav.scp
     save_kaldi_text_mapping(
         data={
-            recording.id: make_wavscp_string(
+            f"{recording.id}_{channel}": make_wavscp_channel_string_map(
```
Comment (pzelasko) [inline lhotse/kaldi.py:221]: I didn’t notice that before but this is possibly breaking existing workflows where data is exported to Kaldi for processing and then imported back to Lhotse (recording ids are changing). I am aware of such use cases. Could you avoid modifying recording ID for single channel recordings?

File: lhotse/kaldi.py [hunk 2]
```diff
@@ -207,26 +209,30 @@ def export_to_kaldi(
...
         },
         path=output_dir / "wav.scp",
     )
     # segments
     save_kaldi_text_mapping(
         data={
-            cut.supervisions[0].id: f"{cut.recording_id} {cut.start} {cut.end}"
+            cut.supervisions[
+                0
+            ].id: f"{cut.recording_id}_{cut.channel} {cut.start} {cut.end}"
```
Comment (pzelasko) [inline lhotse/kaldi.py:235]: Same as above

File: lhotse/kaldi.py [hunk 3]
```diff
@@ -248,7 +254,11 @@ def export_to_kaldi(
     )
     # reco2dur
     save_kaldi_text_mapping(
-        data={recording.id: recording.duration for recording in recordings},
+        data={
+            f"{recording.id}_{channel}": recording.duration
```
Comment (pzelasko) [inline lhotse/kaldi.py:258]: And here

File: lhotse/kaldi.py [hunk 4]
```diff
@@ -248,7 +254,11 @@ def export_to_kaldi(
...
+            for recording in recordings
+            for channel in recording.sources[0].channels
```
Comment (pzelasko) [inline lhotse/kaldi.py:260]: Should probably iterate over sources here too?

Comment (jtrmal) [reply lhotse/kaldi.py:260]: I think we have an assert in the beginning to allow only single-source recordings, no?

File: lhotse/kaldi.py (hunk 1 above)
Comment (jtrmal) [reply lhotse/kaldi.py:221]: will do. Will you take "if" instead of making the list comprehension even more complex? :)

Comment (pzelasko) [reply lhotse/kaldi.py:221]: Yeah. I think at some point this module might need unit tests, it’s definitely growing in complexity. But don’t worry about it for now.

File: lhotse/kaldi.py (hunk 4 above)
Comment (pzelasko) [reply lhotse/kaldi.py:260]: Oh I see. I think your changes are very close to supporting that but it’s fine to leave as-is.

File: lhotse/kaldi.py [hunk 5]
```diff
@@ -194,9 +196,9 @@ def export_to_kaldi(
         "Kaldi export of Recordings with multiple audio sources "
         "is currently not supported."
     )
-    assert all(r.num_channels == 1 for r in recordings), (
-        "Kaldi export of multi-channel Recordings is currently " "not supported."
-    )
+    # assert all(r.num_channels == 1 for r in recordings), (
+    #     "Kaldi export of multi-channel Recordings is currently " "not supported."
+    # )
```
[the remark this answers is not in the document]
Comment (jtrmal) [reply lhotse/kaldi.py]: removed. Sorry

**Comment by jtrmal** [comment]
Actually, I just realized there has to be change to the way how the kaldi wav.scp strings are generated if we want to support full translation identity

**Comment by jtrmal** [comment]
Well, let me know what you think -- it actually relates only to the .sph files

**Comment by pzelasko** [comment]
I'm OK with merging as-is, if we need extra Kaldi compatibility we can follow it up later. Is it OK to merge or did you want to add anything else?

**Comment by pzelasko** [comment]
Thank you, sir!

**Pull request merged**